### ISSUE TITLE
MD030: incorrectly firing inside HTML comments that end with `--->` after switch to micromark

### ISSUE THREAD

**Issue by MarkLodato** [DavidAnson/markdownlint#1067]
As of v0.32.0 (presumably e89ff6ea98a5fbd2477cb0855a8134615f3944c0), MD030 is incorrectly firing inside HTML comments if the comment ends with `--->` instead of `-->`.

Minimal test case:

```markdown
<!--
*  x
--->
```

Steps to reproduce:

```bash
$ mkdir tmp
$ cd tmp
$ cat > test.js <<EOF
const markdownlint = require("markdownlint");
const options = {"files": ["foo.md"]};
console.log(markdownlint.sync(options).toString());
EOF
$ cat > test.md <<EOF
<!--
*  x
--->
EOF
$ npm install markdownlint@v0.31
$ node test.js
$ npm install markdownlint@v0.32
$ node test.js
foo.md: 2: MD030/list-marker-space Spaces after list markers [Expected: 1; Actual: 2]
```

**Comment by DavidAnson** [comment]
An HTML comment that ends with "-" is not really an HTML comment: https://spec.commonmark.org/0.30/#raw-html

**Comment by MarkLodato** [comment]
> An HTML comment that ends with "-" is not really an HTML comment: https://spec.commonmark.org/0.30/#raw-html

Oh, that's crazy! That's much more strict than HTML itself. The additional restrictions were added unexplained in [0.16](https://github.com/commonmark/commonmark-spec/commit/fc75787ce3b927595eb7a5fd324e8bc0ad3c74a1) and is slated to be removed in [0.31](https://github.com/commonmark/commonmark-spec/issues/712) to match HTML

Well, I guess technically this is spec-compliant. :-/

Feel free to close as Working As Intended if you don't plan to fix, though as a heads up this may be an issue once 0.31 gets released.

**Comment by DavidAnson** [comment]
I'd be glad if this were defined more consistently, but for now as behavior is matching specification, I will close the issue. If/when a new specification shows up, I'll try to make corresponding updates.
